libs/designer/src/lib/ui/panel/templatePanel/templatePanel.tsx

```diff
@@ -25,29 +24,24 @@ export const TemplatePanel = ({ createWorkflow, onClose, showCreate, workflowId
   const dispatch = useDispatch<AppDispatch>();
   const intl = useIntl();
   const { selectedTabId, isOpen, currentPanelView } = useSelector((state: RootState) => state.panel);
-  const {
-    templateName,
-    workflowAppName,
-    existingWorkflowName,
-    connections,
-    isConsumption,
-    connectionsError,
-    parametersError,
-    parameterDefinitions,
-  } = useSelector((state: RootState) => ({
+  const { templateName, workflowAppName, manifest, workflows } = useSelector((state: RootState) => ({
     templateName: state.template.templateName,
     workflowAppName: state.workflow.workflowAppName,
-    existingWorkflowName: state.workflow.existingWorkflowName,
-    connections: state.workflow.connections,
-    isConsumption: state.workflow.isConsumption,
-    connectionsError: state.template.errors.connections,
-    parametersError: state.template.errors.parameters,
-    parameterDefinitions: state.template.parameterDefinitions,
+    manifest: state.template.manifest,
+    workflows: state.template.workflows,
   }));
-  const { manifest, workflowName, kind, errors, workflowDefinition } = useWorkflowTemplate(workflowId);
+  const isMultiWorkflowTemplate = useMemo(() => Object.keys(workflows).length > 1, [workflows]);
```

Comment:
Just a consideration, is it better to define it here and pass it around as params? Why are we picking this over storing this at the templateSlice state level?

Reply:
templateSlice already has workflows in its cache, and isMultiWorkflow can be inferred from there, I dont think this is a costly operation and this only gets executed when template panel is being created since this is in memory, so when template card is selected or when use template is selected.

Is your concern about the repeated calculation?

Reply:
My concern was that isMultiWorkflowTemplate is being passed as a param to places such as useCreateWorkflowPanelTabs

Reply:
I was expressing concern for that value being passed as a param when it exists or could exist in the template state

Reply:
But since it's only being passed twice as a param, if you think that's okay, wouldn't be a big deal

Reply:
Lets change it as we need when we are into tabs implementation